src/EPR.SubsidiaryBulkUpload.Application.UnitTests/Services/CompaniesHouseDownload/CompaniesHouseDownloadServiceTests.cs

```diff
@@ -20,76 +23,88 @@ public void TestInitialize()
     }
 
     [TestMethod]
-    public async Task ShouldDownloadAllRequiredFiles()
+    public async Task StartDownload_ShouldDownloadAllRequiredFiles()
     {
         // Arrange
         var month = 3;
         var year = 2024;
         var now = new DateTimeOffset(year, month, 5, 7, 9, 11, TimeSpan.Zero);
         timeProvider.SetUtcNow(now);
+        var partitionKey = now.ToString("yyyyMM");
 
-        var downloadPath = "https://download/";
-
-        fixture.Customize<ApiOptions>(ctx => ctx.With(a => a.CompaniesHouseDataDownloadUrl, downloadPath));
+        fixture.Customize<ApiOptions>(ctx => ctx.With(a => a.CompaniesHouseDataDownloadUrl, DownloadPath));
         var options = fixture.CreateOptions<ApiOptions>();
 
         using var stream = new MemoryStream();
 
-        var partialFileName = $"{downloadPath}{CompaniesHouseDownloadService.PartialFilename}-2024-03-01-part";
-
+        var partialFilePath = $"{DownloadPath}{PartialFilename}-2024-03-01-part";
+        var partialFileName = $"{PartialFilename}-2024-03-01-part";
         var numberOfDownloads = 3;
 
         fixture.Customize<CompaniesHouseFileSetDownloadStatus>(ctx => ctx.With(s => s.CurrentRunExpectedFileCount, numberOfDownloads));
-        var downloadStatus = fixture.Create<CompaniesHouseFileSetDownloadStatus>();
 
         var downloadStatusStorage = new Mock<IDownloadStatusStorage>();
-        downloadStatusStorage.Setup(dss => dss.GetCompaniesHouseFileDownloadStatusAsync()).ReturnsAsync(downloadStatus);
+        downloadStatusStorage.Setup(dss => dss.GetCompaniesHouseFileDownloadStatusAsync(partitionKey)).ReturnsAsync(true);
+
+        var downloadLog = new List<CompaniesHouseFileSetDownloadStatus>();
+        downloadLog.Add(new CompaniesHouseFileSetDownloadStatus { DownloadFileName = $"{partialFileName}1_{numberOfDownloads}.zip" });
```

Comment:
You could use initializer syntax for this, as seen in other methods in this class:
```
  var downloadLog = new List<CompaniesHouseFileSetDownloadStatus>
 {
     new() { DownloadFileName = $"{partialFileName}1_{numberOfDownloads}.zip" },
```  

You can use new() in other places in this class where CompaniesHouseFileSetDownloadStatus is created